src/pieces/rook.cpp

```diff
@@ -71,7 +71,7 @@ std::list<Movement> Rook::get_pseudo_valid_movements(Board& board)
             test_piece = board.get_piece_at(test_coordinate);
             if(test_piece->get_type() == PieceType::king && test_piece->get_had_moved()==false)
             {
-                offset = DirectionOffset.at(Direction::right);
+                offset = DirectionOffset.at(Direction::left);
```

Comment:
Non ho capito perchè deve muoversi ancora a sinistra. Essendo arrocco corto non dovrebbe muoversi di 2 caselle a sinistra dalla sua posizione iniziale ?

Reply:
perdonami ma riscrivi il codice. Fai che la destinazione del movimento sia il re.

Reply:
quindi in pratica la torre dovrebbe andare nella casella del re?

Reply:
sì, avevamo deciso per evitare conflitti con i movimenti "normali"

Reply:
ho già fatto in board

Reply:
Nel mondo reale la torre ovviamente non va nella casella del re, ma qua dobbiamo trovare un modo per far si che non venga confuso con un semplice movimento della torre di due caselle. Quindi o si fa questa mossa speciale, o facciamo che solo il re possa fare l'arrocco.